changelog/unreleased/kong/fix-clustering-forward-proxy-authentication.yml

```diff
@@ -0,0 +1,3 @@
+message: Fixed an issue where hybrid mode not working if the forward proxy password contains special character(#). Note that the `proxy_server` configuration parameter still needs to be url-encoded.
```

Comment:
It's feasible, but the implementation may not be so straightforward. Because if we do this way, there's a caveat: it requires that the user's input is not URL-encoded. If the user's input is already URL-encoded, then it would be encoded twice here, which obviously is incorrect.

To avoid this situation, we need to decode the URL once before parsing it using `url_parse`, ensuring we get plain text. Afterward, we can encode it again for `url_parse` to parse. Personally, I think the process is a bit wordy.

Additionally, `curl` expects users to handle URL encoding themselves in such cases as well:

```
$ curl -x 'http://testuser:rye#!ihef4@localhost:3128' http://httpbin-1/stream/5
curl: (5) Unsupported proxy syntax in 'http://testuser:rye#!ihef4@localhost:3128'
```

Reply:
Make sense. This requires users to provide an encoded string, so should we change the comment in the `kong.conf.default` a bit:

```
#proxy_server =                  # Proxy server defined as an encoded URL. Kong will only use this
```

Reply:
done